python/rascaline/rascaline/utils/clebsch_gordan/_utils.py

```diff
@@ -570,51 +436,27 @@ def cg_tensor_product_blocks_same_samples(
     """
     For a given pair of TensorBlocks and desired angular channels, combines the values
     arrays and returns a new TensorBlock.
+
+    If the samples of the two blocks are different, we assume that one of the sample
+    names are a subset of the other sample names. We then match sample values for
+    this subset, and reshape the block with the smallest sample names to have the
+    same number of samples as the other one.
     """
+    # Reshape the blocks along the samples axis by matching samples dimensions.
+    # TODO: can any reshaping be done on the TensorMap level instead of repeated every
```

Comment:
```suggestion
    # Expand the blocks along the samples axis by matching samples dimensions.
    # TODO: add a new operation in metatensor to do this
```

It's not really a reshape operation